src/workspace.ml

```diff
@@ -186,7 +186,7 @@ let t ?x ?profile:cmdline_profile () =
         Loc.fail (Context.loc ctx)
           "you can only have one context for merlin"
       | Opam { merlin = true; _ }, None ->
-        { merlin_context = Some name; contexts = ctx :: t.contexts; env = None }
+        { t with contexts = ctx :: t.contexts; merlin_context = Some name }
```

Comment:
What do you think about making this more explicit by extracting `set_merlin_context_name` and `add_context` functions?

Reply:
Looking at this code again, it's too complicated for what it is really doing. I simplified it.